addons/l10n_tr/models/account_move_line.py

```diff
@@ -0,0 +1,22 @@
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+class AccountMoveLine(models.Model):
+    _inherit = 'account.move.line'
+
+    l10n_tr_exception_code_id = fields.Many2one(comodel_name='l10n_tr.exception_reason')
+    l10n_tr_exception_code_domain_ids = fields.Many2many(comodel_name='l10n_tr.exception_reason', store=True,
+                                                         compute='_compute_l10n_tr_exception_code_domain_ids')
+
+    @api.depends('tax_ids')
+    def _compute_l10n_tr_exception_code_domain_ids(self):
+        for line in self:
+            line.l10n_tr_exception_code_domain_ids = line.tax_ids.mapped('l10n_tr_exception_code_ids')
+
+    @api.constrains('tax_ids')
+    def _check_tax_compatibility(self):
+        for line in self.filtered_domain([('move_id.country_code', '=', 'TR')]):
+            if len(line.mapped('tax_ids.tax_group_id')) < len(line.tax_ids):
+                raise ValidationError(_(
+                    'You cannot have more than one tax from each tax group in a line. '
```

Comment:
Why? This is not the purpose of the `account.tax.group` feature. It looks like an exploit :p We currently have no generic way to restrict the user to some taxes combination but there are rules in any country. To be consistent, I suggest to completely remove this constraint and let the user does what he want.

Reply:
Yes but the purpose of the tax.group is not that :/ It's purpose is only about display on invoices

Reply:
So I should add another field for this check?